Source/OxyPlot.Wpf/CanvasRenderContext.cs

```diff
@@ -1064,6 +1210,39 @@ private IEnumerable<Point> GetActualPoints(IList<ScreenPoint> points, double str
             }
         }
 
+        /// <summary>
+        /// Snaps points to pixels if required by the edge rendering mode.
+        /// </summary>
+        /// <param name="points">The points.</param>
+        /// <param name="strokeThickness">The stroke thickness.</param>
+        /// <param name="edgeRenderingMode">The edge rendering mode.</param>
+        /// <returns>The processed points.</returns>
+        private Tuple<Point, IList<Point>> GetActualPointsForStreamGeometry(IList<ScreenPoint> points, double strokeThickness, EdgeRenderingMode edgeRenderingMode)
```

Comment:
I can't quite figure out why this function is needed - as far as I can see it does the exact same thing as `GetActualPoints(...)`, except it returns the first point separately. But can't the caller access this via something like `actualPoints[0]` anyway?

Reply:
That's exactly what it does. The issue is not about using actualPoints[0], but that StreamGeometryContext.PolyLineTo takes a an IList<Point> from the second point. Basically there are 2 ways to do it using GetActualPoints() :
- multiple individual StreamGeometryContext.LineTo() calls and either use GetActualPoints().ToList() or count manually to treat the firspoint differently
- Call GetActualPoints().ToList() to get the first point, then use actualPoints.Skip(1).ToList(), which means alllocating an additional array

This implementation allowed me to use the faster PolyLineTo(), while not spending time in ToList(). Not super happy about it, but it's the best solution I could find

Reply:
Hmm, I don't really like the fact that the implementation of `GetActualPoints(...)` is duplicated. Could we use something like this instead:

        static void GetFirstAndRest<T>(IEnumerable<T> sequence, out T first, out IList<T> rest)
        {
            using var e = sequence.GetEnumerator();
            rest = new List<T>();
            if (!e.MoveNext())
            {
                first = default;
                return;
            }

            first = e.Current;
            while (e.MoveNext())
            {
                rest.Add(e.Current);
            }
        }

(untested, just as a concept)